TUNING.md

```diff
@@ -0,0 +1,153 @@
+
+## Tuning At Build-Level
+
+It is most likely that a pre-built binary is being used - either an official one or one provided by the packaging manager of my operating system.
+
+But if you build your own binary it is possible to apply several steps to potentially improve the performance.
+
+Note: Recently more distribution have opted to use more advanced optimization for their packages so several of the following suggestions might have already been applied.
+Please refer to the documentation of your distribution for more insight on this.
+
+### Use Matchcompiler
+
+(TODO: document how to use it when using the in-tree Visual Studio project)
+(TODO: check with a CMake generated Visual Studio project)
+
+### Use Boost
+
+Boost.Container (https://www.boost.org/doc/libs/release/libs/container) is being used for some containers which have a smaller overhead.
+
+As the used library is header-only implementation you only need to install the package on the system you build the binary on but not on the system you run the analysis on.
+
+The official Windows binary is currently not using this (TODO: file ticket).
+
+(TODO: enable usage by default / bail out when it is enforced)
+
+### Use A Different Compiler
+
+Analyzing our own code base has shown that using a different compiler might lead to slightly better performance.
+
+In our case Clang is mostly faster than GCC. See https://trac.cppcheck.net/ticket/10778 for some details.
+
+### Use More Advanced Optimizations
+
+By default we enforce the `-O2` optimization level. Even when using the `Release` build type in CMake which defaults to `-O3`. It might be possible that building with `-O3` might yield a perfomance increase.
+
+There are also no additional code generation flags provided so the resulting binary can run on any system. You might be able to tune this and apply more optimization which is tailored to the system you will be running the binary on.
+
+Needs to be evaluated. (TODO: file tickets)
+
+### Use LTO
+
+Needs to be evaluated. See https://trac.cppcheck.net/ticket/11671.
+
+### Use profile-guided optimizations (PGO/BOLT/AutoFDO/Propeller)
+
+Needs to be evaluated. See https://trac.cppcheck.net/ticket/11672.
+
+## Tuning At Analysis-Level
+
+### Use A More Performant Platform
+
+It seems people assume that you need to run the analysis on the same system you build your code on/for. That is not necessary as the analysis is system-agnostic.
+As system headers are not required for analyzing your code you only need to specify the configuration which matches the system you run your code on.
+
+In case you are using a project as input which can only be generated on the build/target system you can just transfer that to a different system and still run the analysis.
+
+### Specify A Build Dir
+
+Using the `--cppcheck-build-dir` allows you to perform incremental runs which omit files which have not been changed.
```

Comment:
Should probably mention the need to combine `-j` with `--cppcheck-build-dir` for certain checks (`unusedFunction`?)